File: web-frontend/src/domain/usecases/GetAdsUseCase.ts
import { Ad } from '../entities/Ad';
import { IAdRepository } from '../repositories/IAdRepository';

export class GetAdsUseCase {
  constructor(private adRepository: IAdRepository) {}

  async execute(type?: string): Promise<Ad[]> {
    try {
      if (type) {
        return await this.adRepository.getAdsByType(type);
      }
      return await this.adRepository.getAds();
    } catch (error) {
      console.error('Failed to get ads:', error);
      throw new Error('광고 목록을 불러오는데 실패했습니다.');
    }
  }
}

File: android-sdk/samplesdk/src/main/java/com/example/android_sdk/domain/usecases/GetAdsUseCase.kt
package com.example.android_sdk.domain.usecases

import com.example.android_sdk.domain.entities.Ad
import com.example.android_sdk.domain.repositories.AdRepository

class GetAdsUseCase(private val adRepository: AdRepository) {
    suspend fun execute(type: String? = null): List<Ad> {
        return try {
            if (type != null) {
                adRepository.getAdsByType(type)
            } else {
                adRepository.getAds()
            }
        } catch (e: Exception) {
            throw Exception("Failed to get ads: ${e.message}")
        }
    }
}
